### ISSUE TITLE
[Bug]: Published year not displaying on some scraped books if date is encoded as yyyy-mm-dd

### ISSUE THREAD

**Issue by tehguitarist** [advplyr/audiobookshelf#2270]
### Describe the issue

This is something I noticed sometime around v2.4, but I noticed some books didn't have published years, and when they were quick matched, the year didn't populate. When doing a manual match, the year shows up as the year, with some text underneath saying "Currently: yyyy-mm-dd"

It would be good to either have a method to migrate these dates to years, or for ABS to ignore the trailing numbers, given all years should be 4 characters. I suppose an edge case might be mm-dd-yyyy or dd-mm-yyyy, so a test could be used to check if the year is at the front or back, looking for the dashes, or if the 4 characters fall between -5000 to 3000 (to cover first recorded books, into the next thousand years).

### Steps to reproduce the issue

1. Encode the date of a book as yyyy-mm-dd e.g 2013-1-1
2. Observe that there is no date in the Published Date metadata field under Edit Metadata on an audiobook.
3. Observe in manual matching that it is listed as "Currently 2013.1.1

### Audiobookshelf version

2.5.0

### How are you running audiobookshelf?

Docker

**Comment by tehguitarist** [comment]
Upon some investigation, it looks like many tags have Published Year as the published date. I'll see if I can drum up a python script to fix this, if it's not in scope for ABS

**Comment by nichwall** [comment]
Is this similar?
https://github.com/advplyr/audiobookshelf/issues/2258#issuecomment-1783913722

**Comment by tehguitarist** [comment]
I don't think so. Looking at one of the newly created metadata.json files from the recent update I'm seeing this:

  "publishedYear": "2023-01-01",
  "publishedDate": null,

So the fields are just back to front.

**Comment by advplyr** [comment]
Currently published date isn't used anywhere in the UI

**Comment by scapix** [comment]
https://github.com/id3/ID3v2.4/blob/master/id3v2.4.0-changes.txt
 id3v2.4 uses new frames for dates (ones are deprecated), with following formats:
 yyyy, yyyy-MM, yyyy-MM-dd, yyyy-MM-ddTHH, yyyy-MM-ddTHH:mm and  yyyy-MM-ddTHH:mm:ss
same issue happens with mp4/m4a. some tools enforce strict format
e.g. exiftool response to trying to set "Content Creation Date"
```
Warning: Invalid date/time (use YYYY:mm:dd HH:MM:SS[.ss][+/-HH:MM|Z]) in ItemList:ContentCreateDate (PrintConvInv)
Nothing to do.
```

**Comment by LostOnTheLine** [comment]
> Currently published date isn't used anywhere in the UI

Thanks for noting that. I was trying to figure out how to set it in the UI as I'm currently making a script to convert chapter data to `metadata.json` & I was looking at one created by Audiobookshelf & noticed this as one of the fields that was there by default as `null`. I was trying to manually fill every field so that I could identify the Schema so that I can keep consistent with it. I noticed the `  "publishedYear": "2024",` followed by `  "publishedDate": null,`. I tried filling it in with different data, but it seems that it never maters what I enter, it just accepts it

**Comment by blue219** [comment]
I am interested in it, and working on it, thank you
